### ISSUE TITLE
Consul fails to build with Go < 1.5

### ISSUE THREAD

**Issue by talwai** [hashicorp/consul#1288]
As a result of the changes introduced in #1240 , building Consul actually fails on Go < 1.5 because of the format in which `ldflags` are set. The Go docs for link flags [here](https://golang.org/cmd/link/) state the following: 
```
-X importpath.name=value
	Set the value of the string variable in importpath named name to value.
	Note that before Go 1.5 this option took two separate arguments.
	Now it takes one argument split on the first = sign.
```

Since the `importpath.name=value` format is incompatible with Go < 1.5 , `make` actually fails on the build step , simply printing the usage page for `6l` and exiting:
 ```
usage: 6l [options] main.6
...
...
Makefile:7: recipe for target 'all' failed
make: *** [all] Error 2
```

This leads me to assume the linker is unable to correctly parse the flag format. Replacing the `=` sign [here](https://github.com/hashicorp/consul/blob/e34c321fce1bbf0502c74d55448c8e41a7bcb5a3/scripts/build.sh#L49) with a space solves this problem and allows `make` to succeed on Go < 1.5.

If this is a wontfix, I'm happy to build on Go 1.5+ going forward. However the README explicitly states that "(version 1.4+ is _required_)" for building/developing Consul, which is misleading and should be updated.

**Comment by armon** [comment]
@talwai We will fix this for the Consul 0.6 release, but currently build 0.5 against Go 1.4

**Comment by slackpad** [comment]
This is fixed now that we are officially on Go 1.5+.
